tsdb/db.go

```diff
@@ -136,6 +136,11 @@ type Options struct {
 	// It is always the default time and size based retention in Prometheus and
 	// mainly meant for external users who import TSDB.
 	BlocksToDelete BlocksToDeleteFunc
+
+	// MinValidTimeAtInit controls the oldest timestamp we can ingest.
+	// If it is set to 0, we set it to the max timestamp of the oldest compacted
+	// block.
+	MinValidTimeAtInit int64
```

Comment:
I still strongly prefer a function for this, a config option does not smell that good (also think about additional experimentation that it will enable during run time, this config will only help on startup)

Reply:
Addressed it :)